File: solution/452. Minimum Number of Arrows to Burst Balloons/452_01.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](auto& a, auto& b) {
            return a[1] < b[1];
        });

        int res = 1;
        int mx = points[0][1];
        for (auto& p : points) {
            if (p[0] > mx) {
                ++res;
                mx = p[1];
            }
        }
        return res;
    }
};

File: solution/452. Minimum Number of Arrows to Burst Balloons/452_01.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])
        
        res = 1
        arrow_pos = points[0][1]
        
        for x_start, x_end in points:
            if arrow_pos < x_start:
                res += 1
                arrow_pos = x_end
                
        return res
